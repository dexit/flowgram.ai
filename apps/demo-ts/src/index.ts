export * from './types'
export * from './runtime'
export * from './translate'
export * from './validate'
export * from './util'
