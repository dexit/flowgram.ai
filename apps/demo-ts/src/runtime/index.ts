export * from './run'
export * from './RuntimeError'
