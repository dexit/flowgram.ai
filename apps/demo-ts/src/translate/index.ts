export * from './nodes'
export * from './translate'
