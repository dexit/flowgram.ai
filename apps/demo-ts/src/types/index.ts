export * from '@flow-ide-editor/schema'
export * from './nodes'

export class InvalidSchemaError extends Error {}
export class InvalidNodeTypeError extends Error {}
export class InvalidTypeError extends Error {}
export class NotImplementedError extends Error {}

export type FlowArgs = Record<string, unknown>
