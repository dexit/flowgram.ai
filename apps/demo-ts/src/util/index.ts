export * from './log'
export * from './JSBuffer'
