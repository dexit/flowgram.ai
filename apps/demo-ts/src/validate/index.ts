import { FlowNodeSchema } from '@flow-ide-editor/schema'
import { InvalidSchemaError } from '../types'

// TODO use json-schema validation
export function validateOrThrow(node: FlowNodeSchema): void {
  if (!node.definition) throw new InvalidSchemaError()
  if (!node.definition.info) throw new InvalidSchemaError()
  if (!node.definition.info.title) throw new InvalidSchemaError()
}
