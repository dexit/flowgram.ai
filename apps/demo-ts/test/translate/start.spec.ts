import { StartNodeSchema, translateNode } from '../../src'

describe('translate', () => {
  it('Start - 0 param', () => {
    const node: StartNodeSchema = {
      id: 'start',
      type: 'start',
      definition: {
        info: { title: 'start', description: '', icon: '' },
        outputs: {
          type: 'object',
          properties: {},
        },
        script: { language: 'python', content: '' },
      },
      data: {},
    }

    expect(translateNode(node)).toMatchSnapshot()
  })

  it('Start - 1 param', () => {
    const node: StartNodeSchema = {
      id: 'start',
      type: 'start',
      definition: {
        info: { title: 'start', description: '', icon: '' },
        outputs: {
          type: 'object',
          properties: {
            query: { type: 'string', default: 'Hello BlockWise' },
          },
        },
        script: { language: 'python', content: '' },
      },
      data: {},
    }

    expect(translateNode(node)).toMatchSnapshot()
  })

  it('Start - more params', () => {
    const node: StartNodeSchema = {
      id: 'start',
      type: 'start',
      definition: {
        info: { title: 'start', description: '', icon: '' },
        outputs: {
          type: 'object',
          properties: {
            v_string: { type: 'string' },
            v_integer: { type: 'integer' },
            v_number: { type: 'number' },
            v_boolean: { type: 'boolean' },
            v_array: { type: 'array', items: { type: 'string' } },
            v_object: { type: 'object', properties: { a: { type: 'string' } } },
            v_map: {
              type: 'map',
              additionalProperties: { type: 'string' },
            },
          },
        },
        script: { language: 'python', content: '' },
      },
      data: {},
    }

    expect(translateNode(node)).toMatchSnapshot()
  })
})
