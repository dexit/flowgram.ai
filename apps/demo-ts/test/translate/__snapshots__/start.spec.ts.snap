// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`translate Start - 0 param 1`] = `
"  //=== start@start | {level:1,blockIndex:0}
  const start = {}
  start.outputs = {}
  Object.assign(start.outputs, __args)
  "
`;

exports[`translate Start - 1 param 1`] = `
"  //=== start@start | {level:1,blockIndex:0}
  const start = {}
  start.outputs = {
    query: 'Hello BlockWise',
  }
  Object.assign(start.outputs, __args)
  "
`;

exports[`translate Start - more params 1`] = `
"  //=== start@start | {level:1,blockIndex:0}
  const start = {}
  start.outputs = {
    v_string: '',
    v_integer: 0,
    v_number: 0,
    v_boolean: false,
    v_array: [],
    v_object: {
      a: '',
    },
    v_map: {},
  }
  Object.assign(start.outputs, __args)
  "
`;
